src/logging.rs

```diff
@@ -104,6 +108,48 @@ fn is_valid_field(input: &str) -> bool {
     input.chars().all(is_valid_char)
 }
 
+/// A helper for functions that want to take fields as parameters that have already been validated.
+#[derive(Clone, Copy, Debug, Eq, PartialEq)]
+struct ValidField<'a> {
+    field: &'a str,
+}
+
+impl<'a> ValidField<'a> {
+    /// The field value is checked by [[`is_valid_field`]] and a ValidField is returned if true.
+    fn validate(field: &'a str) -> Option<Self> {
+        if is_valid_field(field) {
+            Some(Self { field })
+        } else {
+            None
+        }
+    }
+
+    /// Allows for the construction of a potentially invalid ValidField.  Since
```

Comment:
Minor: if you put an empty line between the first sentence/dot and the rest of the comment, `rustdoc` will properly split into a short one-line summary in the index plus the rest of the description separately:
```rust
/// Allows for the construction of a potentially invalid ValidField.
///
/// Since [...]
```

Reply:
done